FilesStorageKit/Sources/FilesStorageKit/Protocols/FilesStorageProtocol.swift

```diff
@@ -0,0 +1,57 @@
+//
+//  FilesStorageProtocol.swift
+//
+//
+//  Created by Stanislav Jelezoglo on 21.05.2024.
+//
+
+import UIKit
+import CommonKit
+
+public protocol FilesStorageProtocol {
+    func cacheImageToMemoryIfNeeded(id: String, data: Data) -> UIImage?
+    
+    func getPreview(for id: String) -> UIImage?
+    
+    func isCachedLocally(_ id: String) -> Bool
+    
+    func isCachedInMemory(_ id: String) -> Bool
+    
+    func getFileURL(with id: String) throws -> URL
+    
+    func getFile(with id: String) throws -> FilesStorageKit.File
```

Comment:
As I've written before, I don't recommend using throws due to difficulties with error parsing. Do we really need throws here?